mllib/src/test/scala/org/apache/spark/ml/util/MLTestingUtils.scala

```diff
@@ -58,6 +58,30 @@ object MLTestingUtils extends SparkFunSuite {
       "Column label must be of type NumericType but was actually of type StringType"))
   }
 
+  def checkNumericTypesALS[M <: Model[M], T <: Estimator[M]](
```

Comment:
I originally tried to adapt your generic methods to handle the ratings case, but it turned out to seem more complex than was worth it. I kept the generic types as a left-over from that attempt, but yeah it's not really necessary,